Gems/OpenXRVk/Code/Source/XRCameraMovementComponent.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <XRCameraMovementComponent.h>
+
+#include <AzCore/Component/TransformBus.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/RTTI/BehaviorContext.h>
+
+#include <OpenXRVk/InputDeviceXRController.h>
+
+#include <Atom/RPI.Public/ViewProviderBus.h>
+#include <Atom/RPI.Public/View.h>
+#include <AzFramework/Components/CameraBus.h>
+
+#include <Atom/RPI.Public/ViewportContext.h>
+#include <Atom/RPI.Public/ViewportContextBus.h>
+
+
+namespace OpenXRVk
+{
+    void XRCameraMovementComponent::Reflect(AZ::ReflectContext* context)
```

Comment:
Are we using this class within ASV samples for testing? If so, is there another PR for ASV work?

Reply:
I haven't modified ASV sample, but I can test things out there, see if we can integrate this component into it.

Reply:
ASV sample already has some controls working, and they are similar functionality.  If we really want to, we can go do that, but I think for now we can get this through and address that later.